host/CMakeLists.txt

```diff
@@ -136,6 +136,9 @@ if (UNIX)
   list(APPEND PLATFORM_SDK_ONLY_SRC ${PROJECT_SOURCE_DIR}/common/asn1.c
        ${PROJECT_SOURCE_DIR}/common/crypto/openssl/hmac.c
        crypto/openssl/random.c linux/syscall.c)
+  set_source_files_properties(
```

Comment:
Can you add a comment explaining why we need -Wno-deprecated-declarations for common/crypto/openssl/key.c?

Reply:
I see you have it in the PR description "which cannot be upgraded to OpenSSL 3 due to ENGINE not being supported by OpenSSL 3." I think it would be nice to have as a comment in this file too.